lib/pages/edit_task.dart

```diff
@@ -3,65 +3,49 @@ import 'package:go_router/go_router.dart';
-
-  @override
-  State<EditTask> createState() => _EditTaskState();
-}
+  final Function(String) onTaskUpdated;
 
-class _EditTaskState extends State<EditTask> {
-  late TextEditingController _controller;
-  @override
-  void initState() {
-    super.initState();
-    _controller = TextEditingController(text: widget.task.name);
-  }
+  const EditTask({super.key, required this.task, required this.onTaskUpdated});
 
   @override
   Widget build(BuildContext context) {
+    TextEditingController controller = TextEditingController(text: task.name);
+
     return Form(
-        key: const Key('Form'),
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.start,
-          children: [
-            Padding(
-              padding: const EdgeInsets.fromLTRB(0, 0, 0, 42),
-              child: TextField(
-                style: TextStyle(color: iconColor),
-                controller: _controller,
-                decoration: InputDecoration(
-                  labelText: 'Edit Task Name',
-                  labelStyle: TextStyle(color: white, fontSize: 20),
-                ),
+      key: const Key('edit_task_form'),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: [
+          Padding(
+            padding: const EdgeInsets.only(bottom: 42),
+            child: TextField(
+              style: TextStyle(color: iconColor),
+              controller: controller,
+              decoration: InputDecoration(
+                labelText: 'Edit Task Name',
+                labelStyle: TextStyle(color: white, fontSize: 20),
               ),
             ),
-            ElevatedButton(
-              style: ElevatedButton.styleFrom(
-                backgroundColor: black,
-              ),
-              onPressed: () {
-                // Get the updated task from the TextField
-                final String updatedTaskName = _controller.text;
-
-                // Create a new TaskName object with the updated name
-                final Task updatedTask =
-                    Task(id: widget.task.id, name: updatedTaskName);
-
-                // Call the callback function to return the updated task
-                widget.onTaskUpdated(updatedTask);
-
-                // Close the edit panel
-                context.pop(context);
-              },
-              child: Text(
-                'Save',
-                style: TextStyle(color: white, fontSize: 20),
-              ),
+          ),
+          ElevatedButton(
+            key: const Key('edit_task_button'),
+            style: ElevatedButton.styleFrom(
+              backgroundColor: black,
+            ),
+            onPressed: () {
+              final String updatedTaskName = controller.text;
+              onTaskUpdated(updatedTaskName);
+              context.pop(context);
+            },
+            child: Text(
+              'Save',
+              style: TextStyle(color: white, fontSize: 20),
```

Comment:
Consider saving this `TextStyle` as a class-level variable since it is used twice.

Reply:
done